src/ewatercycle/observation/grdc.py

```diff
@@ -104,23 +104,20 @@ def get_grdc_data(
     station_not_in_nc = False
     if nc_file.exists():
         ds = xr.open_dataset(nc_file)
-        try:
+        if int(station_id) in ds["id"]:
             ds = ds.sel(
                 id=int(station_id),
                 time=slice(get_time(start_time).date(), get_time(end_time).date()),
             )
-            ds = ds.rename({"runoff_mean": column})
-            if len(ds.time) != 0:
-                return ds
-        except KeyError:
-            station_not_in_nc = True
+            return ds.rename({"runoff_mean": column})
+        station_not_in_nc = True
```

Comment:
This variable isn't really needed anymore right?

Reply:
to give a better error at line 119 it is still needed.